ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStoreManager.java

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.cloud.azure.AzureCloudConfig;
+import com.github.ambry.cloud.azure.AzureMetrics;
+import com.github.ambry.cloud.azure.ConnectionStringBasedStorageClient;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Storage manager that tracks {@link AzureStore} objects.
+ */
+public class AzureStoreManager implements StoreManager {
```

Comment:
Same as earlier comment in AzureStore class. May be we could rename the classes under `/ambry/cloud` to `{oldName}V2` such `CloudStorageManagerV2` and have Azure names for classes only under `/ambry/cloud/azure`?

Reply:
I highly doubt we will ever depart from Azure, so just using the name `Azure` directly is descriptive. Giving new classes a distinct prefix makes searching easier with grep or any code browser. For eg 
```
find . -name 'AzureStore'
```
will show up all the V2 code and not V1 code.